src/main/kotlin/io/github/gradlenexus/publishplugin/CloseNexusStagingRepository.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import io.github.gradlenexus.publishplugin.internal.NexusClient
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.property
+import javax.inject.Inject
+
+@Suppress("UnstableApiUsage")
+open class CloseNexusStagingRepository @Inject
+constructor(objects: ObjectFactory, extension: NexusPublishExtension, repository: NexusRepository) :
+        BaseOperationOnNexusStagingRepository(objects, extension, repository) {
+
+    @get:Input
+    val stagingRepositoryId: Property<String> = objects.property()
```

Comment:
Would it make sense to allow passing this as an `@Option`, too?

Reply:
Hmm, what would be the behavior in that case? Close all repositories?

Reply:
No, the option would allow passing the stagingRepositoryId as a task option on the command line, e.g. `closeSonatypeStagingRepository --staging-repo-id=comacme-1234`

Reply:
Sure, I confused `Option` with `Optional` :man_facepalming: 

`Option` is definitely a way to go :).

Reply:
Done.